### ISSUE TITLE
Retries are disappearing with Maxretry handler

### ISSUE THREAD

**Issue by jagthedrummer** [jondot/sneakers#183]
I'm trying to follow the docs on using the Maxretry handler, and it looks like the handler is trying to retry, but nothing is showing up in the retry queue.

Here's the error I'm seeing in the worker console:

```
... ERROR: [worker-development.marketplace.purchase.created.charge:1:bowvwu][#<Thread:0x007fed2e1d6ff0>][development.marketplace.purchase.created.charge][#<Sneakers::Configuration:0x007fed2e1dd648>] unexpected error [Exception error="Couldn't find Purchase with 'id'=77" error_class=ActiveRecord::RecordNotFound backtrace=/Users/jgreen/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.3/lib/active_record/core.rb:155:in `find',/Users/jgreen/projects/clickfunnels/clickfunnels-marketplace-api/app/workers/purchase_charger_worker.rb:18:in `work_with_params',/Users/jgreen/.rvm/gems/ruby-2.2.2/bundler/gems/sneakers-e86de84fef2a/lib/sneakers/worker.rb:56:in `block (3 levels) in do_work',/Users/jgreen/.rvm/gems/ruby-2.2.2/bundler/gems/sneakers-e86de84fef2a/lib/sneakers/metrics/null_metrics.rb:8:in `call',/Users/jgreen/.rvm/gems/ruby-2.2.2/bundler/gems/sneakers-e86de84fef2a/lib/sneakers/metrics/null_metrics.rb:8:in `timing',/Users/jgreen/.rvm/gems/ruby-2.2.2/bundler/gems/sneakers-e86de84fef2a/lib/sneakers/worker.rb:54:in `block (2 levels) in do_work',/Users/jgreen/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/timeout.rb:89:in `block in timeout',/Users/jgreen/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/timeout.rb:99:in `call',/Users/jgreen/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/timeout.rb:99:in `timeout',/Users/jgreen/.rvm/gems/ruby-2.2.2/bundler/gems/sneakers-e86de84fef2a/lib/sneakers/worker.rb:53:in `block in do_work',/Users/jgreen/.rvm/gems/ruby-2.2.2/gems/thread-0.2.2/lib/thread/pool.rb:67:in `call',/Users/jgreen/.rvm/gems/ruby-2.2.2/gems/thread-0.2.2/lib/thread/pool.rb:67:in `execute',/Users/jgreen/.rvm/gems/ruby-2.2.2/gems/thread-0.2.2/lib/thread/pool.rb:405:in `block (2 levels) in spawn_thread',/Users/jgreen/.rvm/gems/ruby-2.2.2/gems/thread-0.2.2/lib/thread/pool.rb:372:in `loop',/Users/jgreen/.rvm/gems/ruby-2.2.2/gems/thread-0.2.2/lib/thread/pool.rb:372:in `block in spawn_thread']
... INFO: Maxretry handler [queue=development.marketplace.purchase.created.charge] msg=retrying, count=1, headers=
```

Here's my sneakers initializer (in a Rails project):
```ruby
require 'sneakers'
require 'sneakers/handlers/maxretry'
Sneakers.configure :amqp => ENV["RABBITMQ_URL"],
                   :heartbeat => 5,
                   :workers => 1,
                   :prefetch => 1,
                   :threads => 1,
                   :timeout_job_after => 30,
                   :handler => Sneakers::Handlers::Maxretry,
                   :retry_timeout => 50000,
                   :retry_max_times => 5,
                   :ack => true # must acknowledge or msgs will be retried
```

Any my worker:

```ruby
class PurchaseChargerWorker
  include Sneakers::Worker
  queue_name = "#{Rails.env}.marketplace.purchase.created.charge"
  from_queue queue_name,
    :exchange => "#{Rails.env}.marketplace.purchase.created",
    :exchange_options => { :durable => true, :type => :fanout },
    :durable => true,
    :ack => true,
    :arguments => {
      :'x-dead-letter-exchange' => "#{queue_name}.retry"
    }

  def work_with_params(msg,delivery_info,metadata)
    json = JSON.parse(msg)
    puts "json = #{json}"
    puts "delivery_info = #{delivery_info}"
    puts "metadata = #{metadata}"
    purchase = Purchase.find json["purchase_id"]
    PurchaseCharger.new.charge(purchase)
    ack!
  end
end
```

I can see that the DLX exchanges and queues are being created, but it seems like the retry just isn't happening. What am I missing?

**Comment by gabrieljoelc** [comment]
What version of sneakers are you using? If you using 2.3.x, then you need to use `:queue_options => { :arguments => { :'x-dead-letter-exchange' => "#{queue_name}.retry" } }`

**Comment by jagthedrummer** [comment]
Hmm... I'm using 2.3.4. When I move the DLX `:arguments` into `:queue_options` I get the same thing. It looks like Sneakers reports that it should retry, but the message just disappears.

```
INFO: Maxretry handler [queue=development.marketplace.purchase.created.charge] msg=retrying, count=1, headers=
```

**Comment by jagthedrummer** [comment]
I should note that I'm seeing the retry/error queues/exchanges being created, they just don't ever get a message.

**Comment by jagthedrummer** [comment]
OK, I finally figured this out. The issue was that I was setting the `:'x-dead-letter-exchange'` value to be `"#{queue_name}.retry"` (with a dot) instead of `"#{queue_name}-retry"`. I was under the impression that I was giving Sneakers the name of the retry exchange I wanted it to use AND create. Turns out that Sneakers will just create a retry exchange with `whatever_queue_name_the_worker_is_using + "-retry"`. It's always `"-retry"` no matter what you specify in the `:'x-dead-letter-exchange'` value. It's then up to you as the developer to properly match up the value for `:'x-dead-letter-exchange'` to be exactly what Sneakers auto creates for you. 

So, after figuring that out I have two suggestions:

1. I think the docs could be clearer that the value of `:'x-dead-letter-exchange'` has strict requirements and that it is not the place where you tell Sneakers what you WANT the retry exchange to be named, but rather it's where to tell Sneakers something it should already know.

2. It would be great if Sneakers would set that up automatically when you're using the Maxretry handler.
